### DISCUSSION TITLE
STL import not working well

### DISCUSSION THREAD

**Discussion by rajpiyush18** [ankrh/MCmatlab#4] [Q&A] [answered]
I am working with the following file 
[angled_square_needle_v1.zip](https://github.com/ankrh/MCmatlab/files/9926131/angled_square_needle_v1.zip)

I am using the following command: 

theta = -pi/2;
  A = 0.1*[1     0            0    ;
           0 cos(theta) -sin(theta);
           0 sin(theta)  cos(theta)]; % Rotation around the x axis with -pi/2

% and then translation
  v = [-0.1 -0.025 3.108];

It is working when
v = [_positive x value_, -0.025 3.108];

The moment I put a negative value for x, it doesn't seem to work

Other details:

model.G.nx                = 200; % Number of bins in the x direction
model.G.ny                = 200; % Number of bins in the y direction
model.G.nz                = 200; % Number of bins in the z direction
model.G.Lx                = 0.8; % [cm] x size of simulation cuboid
model.G.Ly                = 0.8; % [cm] y size of simulation cuboid
model.G.Lz                = 0.8; % [cm] z size of simulation cuboid

**Comment by ankrh** [comment, marked as answer]
Thanks for the bug report! I managed to track down the bug to the function VOXELIZE, which I originally got from the Mathworks FileExchange. I found a fork of it, written by W. Warriner, who fixed this bug and many other improvements. I will replace the VOXELIZE function in the next release of MCmatlab, but in the meanwhile if you put these files into the helperfuncs folder it should fix the issue:

[Unzip me into helperfuncs.zip](https://github.com/ankrh/MCmatlab/files/9930773/Unzip.me.into.helperfuncs.zip)

Let me know if it works or not.

**Reply by rajpiyush18** [reply]
Thank you so much @ankrh. This works very nicely.
